app/src/main/java/azul/TileHolder.java

```diff
@@ -5,19 +5,48 @@
 import java.util.Optional;
 import java.util.stream.Collectors;
 
-abstract class TileHolder {
+public class TileHolder {
     protected final List<Tile> tiles;
+    private final String identifier;
+    private static List<TileHolder> holders = new ArrayList<TileHolder>();
+
+
 
     protected TileHolder() {
         tiles = new ArrayList<>();
+        var id = holders.stream()
+                .filter(h -> h.getClass().getName().equals(this.getClass().getName())).count() + 1;
+
+        identifier = this.getClass().getName() + id;
+    }
+
+
+    /**
+     * 
+     * 
+     * @return The generated identifier for this TileHolder.
+     */
+    public String getIdentifier() {
+        return identifier;
```

Comment:
Miért van szükség ID-ra?

Reply:
A parancsoros felület miatt célszerű, hogy pontos névvel ki lehessen választani user input alapján melyik `TileHolder`-ről emelünk le dolgokat.
Lesz maga egy játék logika, ami függeni fog egy [AzulUI](https://github.com/Uni-Sopron/szta-22t-azul-azure/blob/main/app/src/main/java/azul/interfaces/AzulUI.java)-tól.
A konkrét parancssoros megvalósítása a [UIInputParser](https://github.com/Uni-Sopron/szta-22t-azul-azure/blob/main/app/src/main/java/azul/UIInputParser.java) (ezt majd lehet célszerűbb lenne átnevezni)
Ennek a `getTileHolderId()` függvénye visszaadja nyers stringként a helyes ID-t, a `TileHolder`-be pedig még belekerül egy `TileHolder parseFromStringId()` jellegű függvény ami visszadobja a belső `holders` tárolóból a megfelelő `TileHolder` objektumot.
Azért is a `TileHolder`-be került ez, mert így a kódrészletnél, ahol a kiválasztott helyről leemelünk `Tile`-kat, nem kell tudni a pontos osztályt (`ManufactureDisk` vagy `Middle`), a `pop()` helyesen  felülírt változata elég.